### PR TITLE
[python] Fix python api can't connect to remote gateway server

### PR DESCRIPTION
**Pull request by zhongjiajie**
For now, python API could only communicate python gateway server
in the same hosts, this patch makes it could work with different hosts,
and export java gateway setting to configure file

### CODE REVIEW

**Comment by zhongjiajie** [comment]
I've already test in my server and it work.

File: dolphinscheduler-python/src/main/java/org/apache/dolphinscheduler/server/PythonGatewayServer.java [hunk 1]
```diff
@@ -501,10 +507,28 @@ public void createUser(String userName,
...
+                null
+            );
+            GatewayServer.turnLoggingOn();
+            logger.info("PythonGatewayServer started on: " + gatewayHost.toString());
+            server.start();
+        }
+        catch (UnknownHostException e) {
+            logger.error("exception occurred while constructing PythonGatewayServer().");
+            e.printStackTrace();
+        }
```
Comment (kezhenxu94) [inline dolphinscheduler-python/src/main/java/org/apache/dolphinscheduler/server/PythonGatewayServer.java]:
```suggestion
        } catch (UnknownHostException e) {
            logger.error("exception occurred while constructing PythonGatewayServer().", e);
        }
```

**Comment by kezhenxu94** [comment]
Others LGTM

File: dolphinscheduler-python/src/main/java/org/apache/dolphinscheduler/server/config/PythonGatewayConfig.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+ */
+
+package org.apache.dolphinscheduler.server.config;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.stereotype.Component;
+
+@Component
```
Comment (ruanwenjun) [inline dolphinscheduler-python/src/main/java/org/apache/dolphinscheduler/server/config/PythonGatewayConfig.java:25]: The `@Configuration` annotation already contains `@Component`, so maybe we don't need to add `@Conponent` here?

File: dolphinscheduler-python/src/main/resources/application.yaml [hunk 3]
```diff
@@ -37,6 +37,22 @@ spring:
...
+  gateway-server-address: 0.0.0.0
+  # The port of Python gateway server start. Define which port you could connect to Python gateway server from
+  # Python API side.
+  gateway-server-port: 25333
+  # The address of Python callback client.
+  python-address: 127.0.0.1
+  # The port of Python callback client.
+  python-port: 25334
+  # Close connection of socket server if no other request accept after x milliseconds.
+  connect-timeout: 0
```
Comment (ruanwenjun) [inline dolphinscheduler-python/src/main/resources/application.yaml:53]: It's better to introduce at here that `0 means infinite`, I am worried user don't know what 0 means.

File: dolphinscheduler-python/src/main/java/org/apache/dolphinscheduler/server/PythonGatewayServer.java (hunk 1 above)
Comment (zhongjiajie) [reply dolphinscheduler-python/src/main/java/org/apache/dolphinscheduler/server/PythonGatewayServer.java]: Thanks

File: dolphinscheduler-python/src/main/java/org/apache/dolphinscheduler/server/config/PythonGatewayConfig.java (hunk 2 above)
Comment (zhongjiajie) [reply dolphinscheduler-python/src/main/java/org/apache/dolphinscheduler/server/config/PythonGatewayConfig.java:25]: Will remove it

File: dolphinscheduler-python/src/main/resources/application.yaml (hunk 3 above)
Comment (zhongjiajie) [reply dolphinscheduler-python/src/main/resources/application.yaml:53]: thanks, I would add more detail comment on it.

**Comment by zhongjiajie** [comment]
Changed base on some suggestions, PTAL @kezhenxu94 @ruanwenjun thanks

**Review by ruanwenjun — APPROVED** [review]
LGTM

**Comment by zhongjiajie** [comment]
Thanks ~

**Pull request merged**